web/src/reducers/index.js

```diff
@@ -1,19 +1,22 @@
 import { routerReducer } from 'react-router-redux';
-import { combineReducers } from 'redux';
 
 import activities from './activities';
 import apd from './apd';
 import auth from './auth';
+import budget from './budget';
 import state from './state';
 import user from './user';
 
-const rootReducer = combineReducers({
-  activities,
-  apd,
-  auth,
-  state,
-  user,
-  router: routerReducer
-});
-
-export default rootReducer;
+export default (oldState = {}, action) => {
+  const newState = {
+    activities: activities(oldState.activities, action),
+    apd: apd(oldState.apd, action),
+    auth: auth(oldState.auth, action),
+    budget: {},
+    state: state(oldState.state, action),
+    user: user(oldState.user, action),
+    router: routerReducer(oldState.router, action)
+  };
+  newState.budget = budget(oldState.budget, action, newState);
+  return newState;
+};
```

Comment:
I don't ***love*** this, but in order to make the budget reducer useful, it needs access to the `activities` state.  `combineReducers` doesn't allow for that.  But it's not hard to mimic what `combineReducers` does, so I did that.  I'd hoped I could just call `rootReducer(state, action)`, but redux actually throws out any state properties that aren't defined in `combineReducers`, so budget stuff was getting lost and we were getting a console error.

Reply:
i don't love this either, though i do understand why we're doing it.

this is basically following bullet 2 from this FAQ: https://redux.js.org/faq/reducers#how-do-i-share-state-between-two-reducers-do-i-have-to-use-combinereducers 

i'm curious for your thoughts on bullets 1 and 3:
* 1 would basically be having `budget` be nested under the `activities` reducer
* 3 would be augmenting our `updateActivity` action and adding `thunk` here so that we can add the activities state into the action payload if `isExpense = true`

option 3 sounds kinda appealing to me in that we keep the budget reducer separate and don't have to change the combineReducer but I'd love your thoughts.

Reply:
🤔 It occurs to me there's another issue at play here that I forgot about when I wrote that earlier text - the `budgets` reducer relies on the updated `activities` state - that is, the `activities` reducer must run first, and its output is fed into the `budgets` reducer.  It's not great.

I do like bullet 3 though.  What do you think about doing something like this?

- `updateActivity` is called
- it dispatches an `UPDATE_ACTIVITY` action; the budget reducer no longer response to this action
- using thunkification, we wait for that action to finish
- if `isExpense` is true, dispatch an `UPDATE_BUDGET` action; the budget reducer responds

Then we get `combineReducers` back and we don't have to worry about the order reducers are run in response to an action - there's an explicit action for updating the budget data.

Reply:
Did that and it's ***SO MUCH CLEANER***.  Thanks for this feedback!

Reply:
awesome! this is great 🥇